weblate/trans/models/translation.py

```diff
@@ -1390,6 +1392,8 @@ def remove(self, user: User) -> None:
             author=user,
         )
 
+        cleanup_stale_glossaries.delay(self.component.project.id)
```

Comment:
The cleanup should be triggered for non-glossary components to avoid calling it again when removing blank glosssaries:

```suggestion
        if not self.component.is_glossary:
            cleanup_stale_glossaries.delay(self.component.project.id)
```